settings.php

```diff
@@ -36,8 +42,88 @@
         get_string('menulastrefresh', 'local_envbar'),
         new moodle_url('/local/envbar/last_refresh.php'));
 
+    $presentation = new admin_settingpage('local_envbar_presentation',
+            get_string('menupresentation', 'local_envbar'));
+
+    $presentation->add(new admin_setting_heading('local_envbar/envbarheading',
+            get_string('envbarheading', 'local_envbar'),
+            null));
+
+    $presentation->add(new admin_setting_configtextarea('local_envbar/extracss',
+            get_string('extracss', 'local_envbar'),
+            get_string('extracss_desc', 'local_envbar'),
+            envbarlib::get_default_extra_css(),
+            PARAM_RAW,
+            50,
+            10));
+
+    $presentation->add(new admin_setting_configtext('local_envbar/stringseparator',
+            get_string('stringseparator', 'local_envbar'),
+            get_string('stringseparator_desc', 'local_envbar'),
+            '-',
+            PARAM_CLEANHTML));
+
+    $presentation->add(new admin_setting_configcheckbox('local_envbar/showconfiglink',
+            get_string('showconfiglink', 'local_envbar'),
+            get_string('showconfiglink_desc', 'local_envbar'),
+            true));
+
+    $presentation->add(new admin_setting_heading('local_envbar/menuheading',
+            get_string('menuheading', 'local_envbar'),
+            null));
```

Comment:
admin_setting_heading expects 3rd parameter to be string. Looks like it doesn't cause any issues now, but potentially in could be broken if related code will be changed.

Reply:
Sure, this is a mistake by me. Sorry for that.
I have fixed it by passing an empty string instead of null now.
I have also added lazy loading of strings to all strings on settings.php which might be benefical.